### DISCUSSION TITLE
86. 트랜잭션과 ACID에 대해서 설명해 주세요.

### DISCUSSION THREAD

**Discussion by cutehumanS2** [jewelry-box/cs-interview#88] [General]
트랜잭션과 ACID에 대해서 설명해 주세요.

**Comment by linirini** [comment]
트랜잭션이란 데이터베이스의 상태를 변화시키기 위해 수행하는 하나의 논리적인 작업 단위이다.
### 트랜잭션 특징
- **원자성(Atomicity)**
    - 트랜잭션의 모든 연산들은 정상적으로 수행 완료되거나 아니면 전혀 어떠한 연산도 수행되지 않은 상태를 보장해야 한다.
- **일관성(Consistency)**
    - 어떤 트랜잭션이 수행되기 전에 데이터베이스가 일관된 상태를 가졌다면 트랜잭션 완료 후에도 데이터베이스가 일관된 상태로 유지되어야 한다.
- **격리성(Isolation)**
    - 하나의 트랜잭션이 실행하는 도중에 변경한 데이터는 이 트랜잭션이 완료될 때까지 다른 트랜잭션이 참조하지 못한다.
- **지속성(Durability)**
    - 트랜잭션이 성공적으로 완료되었으면, 결과는 영구적으로 반영되어야 한다.

하나의 트랜잭션은 Commit 되거나 Rollback 된다.
